bin/pasta_prepare_evaluation.py

```diff
@@ -343,33 +353,229 @@ def _load_responses_dict(msg_id, response_list):
+
+    # Consider only relevant patches (as per given definition of relevance)
+    relevant_patches = get_relevant_patches(patch_characteristics)
+    final_filtered_1 = final[final['patch_id'].isin(relevant_patches)]
+
+    # Filter responses -- only responses to the patch itself count as a response, and not the rest of the thread emails
+    final_filtered_2 = final_filtered_1[final_filtered_1['patch_id'] == final_filtered_1['responses.parent']]
+
+    global _repo
+    _repo = repo
+
+    p1 = Pool(processes=int(cpu_count()), maxtasksperchild=1)
+    response_to_bot = p1.map(_is_response_from_bot, list(final_filtered_2['responses.resp_msg_id'].unique().compute()),
+                            chunksize=1000)
+    p1.close()
+    p1.join()
+
+    _repo = None
+
+    response_bot_df = pd.DataFrame(response_to_bot, columns=['responses.resp_msg_id', 'response_is_bot'])
+
+    final_filtered_2 = dd.merge(final_filtered_2, response_bot_df, how='left', on=['responses.resp_msg_id'])
+
+    if 'response_is_bot_x' in final_filtered_2.columns:
+        final_filtered_2 = final_filtered_2.drop(['response_is_bot_x'], axis=1) \
+            .rename(columns={"response_is_bot_y": "response_is_bot"})
+
+    # Filter out responses from bots
+    final_filtered_3 = final_filtered_2[final_filtered_2['response_is_bot'] != True]
+
+    final_filtered_3.to_csv(config.f_filtered_responses, single_file=True)
+
+    log.info("Written filtered response dataframe to disk, Done!")
+
+
 def prepare_evaluation(config, argv):
     parser = argparse.ArgumentParser(prog='prepare_evaluation',
                                      description='aggregate commit and patch info')
 
     parser.add_argument('--ignored',
-                        action='store_const',
-                        const='ignored',
-                        dest='mode',
+                        action='store_true',
+                        default=False,
                         help='prepare data for patch analysis \n'
                              'prepare data for ignored patch analysis \n'
                         )
 
-    parser.add_argument('--off-list',
-                        action='store_const',
-                        const='off-list',
-                        dest='mode',
+    parser.add_argument('--offlist',
+                        action='store_true',
+                        default=False,
                         help='prepare data for off-list patch analysis \n')
 
     parser.add_argument('--review',
-                        action='store_const',
-                        const='review',
-                        dest='mode',
+                        default=None,
+                        choices=['prepare', 'preprocess', 'merge', 'filter', 'analyze'],
                         help='prepare data for patch review analysis \n')
 
     analysis_option = parser.parse_args(argv)
 
-    if not analysis_option.mode:
+    if len(argv) == 0:
```

Comment:
... Then this is implicitly solved, as the parser will fail in case we didn't provide a choice.